app/jobs/auth_lookup_maintenance_job.rb

```diff
@@ -0,0 +1,39 @@
+class AuthLookupMaintenanceJob < ApplicationJob
+  RUN_PERIOD = 8.hours.freeze
+
+  queue_as QueueNames::AUTH_LOOKUP
+  queue_with_priority  3
+
+  def perform
+    check_authlookup_consistency
+  end
+
+  # checks lookup_table_consistent? on each type for each user, and if not triggers a job to repopulate for that user
+  def check_authlookup_consistency
+    found_types = [].to_set
+
+    Seek::Util.authorized_types.each do |type|
+      #skip if there are any items of this type queued
+      next if AuthLookupUpdateQueue.where(item_type: type.name).any?
+
+      items_for_queue = [].to_set
+
+      User.where.not(person_id: nil).to_a.push(nil).each do |user|
```

Comment:
I did try running the `RegularMaintenceJob ` yesterday whilst the tables were in a stable state, meaning it looped over every type and user checking `lookup_table_consistent?` and it finished in a couple of minutes maximum. 
The additional overhead of getting the `items_missing_from_authlookup` might potentially cause a timeout if things are in a very bad state, but I think this would be unusual and if so it will continue where it left off next time it runs.